stdlib/public/core/ContiguousArrayBuffer.swift

```diff
@@ -750,9 +750,13 @@ internal struct _UnsafePartiallyInitializedContiguousArrayBuffer<Element> {
         _uninitializedCount: newCapacity, minimumCapacity: 0)
       p = newResult.firstElementAddress + result.capacity
       remainingCapacity = newResult.capacity - result.capacity
-      newResult.firstElementAddress.moveInitialize(
-        from: result.firstElementAddress, count: result.capacity)
-      result.count = 0
+      if !result.isEmpty {
```

Comment:
The change looks fine, I'm just probably overthinking the weird ways this condition gets to basically saying `result.count != 0`, and it's effects on performance and clarity of what's going on =)